server/src/application/application.service.ts

```diff
@@ -100,14 +102,42 @@ export class ApplicationService {
   }
 
   public async searchApplications(searchParams: SearchApplicationDto) {
-    return await this.prisma.application.findMany({
-      where: {
-        label: {
-          contains: searchParams.label,
-          mode: 'insensitive',
+    const { label } = searchParams;
+
+    const whereClause: Prisma.ApplicationWhereInput = {
+      ...(label
+        ? {
+            label: {
+              contains: label,
+              mode: Prisma.QueryMode.insensitive,
+            },
+          }
+        : {}),
+    };
+
+    const orderByClause: Prisma.Enumerable<Prisma.ApplicationOrderByWithRelationInput> =
+      {
+        metadata: {
+          updatedAt: 'desc',
         },
-      },
-    });
+      };
+
+    try {
+      const applications = await this.prisma.application.findMany({
+        where: whereClause,
+        orderBy: orderByClause,
+        take: 12,
+        include: {
+          lifecycle: true,
+          actors: true,
+          metadata: true,
+        },
+      });
+
+      return applications;
+    } catch (error) {
+      throw new InternalServerErrorException('Erreur interne du serveur.');
```

Comment:
Pas besoin de gérer les internal errors => https://docs.nestjs.com/exception-filters

Reply:
ok

Reply:
Exemple : https://swagger.io/resources/articles/best-practices-in-api-design/